### PR TITLE
migrate honeycombio_trigger resource 'as is' to Plugin Framework

### PR DESCRIPTION
**Pull request by jharley**
First pass at migrating the resource: with this (verbose) groundwork laid the test suite can be fleshed out and the open bugs squashed 🤞🏻 

The current SDKv2 based Trigger resource is being left in place (and its tests skipped) just not wired up for use.

### CODE REVIEW

File: internal/helper/validation/divisible_by.go
Comment (jharley) [inline internal/helper/validation/divisible_by.go:1]: The Framework didn't get this validator included which is present in SDKv2.

Comment (jharley) [inline internal/helper/validation/divisible_by.go:1]: SDKv2 included such a validator, and we used it, but the Plugin Framework does not. This is a reimplementation.

File: honeycombio/resource_trigger_test.go
Comment (jharley) [inline honeycombio/resource_trigger_test.go:1]: With the Trigger resource removed from the SDKv2 provider, these tests won't run (they can't find the resource). Leaving them here until all the code is removed, so just skipping for easy-mode.

**Review by brookesargent — APPROVED** [review]
🏆

File: internal/provider/trigger_resource.go [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+func (r *triggerResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_trigger"
+}
+
+func (r *triggerResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages a Trigger.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Numeric identifier of the Trigger.",
```
Comment (dstrelau) [inline internal/provider/trigger_resource.go:53]: Numeric?

Comment (jharley) [reply internal/provider/trigger_resource.go:53]: 🙃 I think that was boilerplate copy-pasta. Thanks!

**Pull request merged**